### PR TITLE
feat: Add split button component

### PR DESCRIPTION
**Pull request by JKMarkowski**
#### Please provide a link to the associated issue.
fixes https://github.com/SAP/fundamental-ngx/issues/949
fixes https://github.com/SAP/fundamental-ngx/issues/652
#### Please provide a brief summary of this pull request.
I added split button component.
#### If this is a new feature, have you updated the documentation?
Yes there is completely new page with documentation about split button and some examples.

### CODE REVIEW

**Comment by droshev** [comment]
@JKMarkowski could you resolve the conflicts?

**Comment by JKMarkowski** [comment]
Yes, conflicts resolved.

**Comment by droshev** [comment]
@MattL75 @mikerodonnell89

File: docs/app/documentation/component-docs/button-split/button-split-header/button-split-header.component.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<header>Button</header>
```
Comment (MattL75) [inline docs/app/documentation/component-docs/button-split/button-split-header/button-split-header.component.html]: Change to "Split Button", this also applies to every instance where "Button Split" is used. Fundamental-Styles calls it Split Button so we should follow their lead on this.

File: docs/app/documentation/component-docs/button-split/examples/button-split-examples.component.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export class ButtonSplitTypesIconsComponent {num = 0;}
+
+@Component({
+    selector: 'fd-button-split-programmatical-example',
+    templateUrl: './button-split-programmatical-example.component.html',
+    styleUrls: ['./button-split-examples.scss']
+})
+export class ButtonSplitProgrammaticalExampleComponent {
+    isOpen: boolean = false;
+}
```
Comment (MattL75) [inline docs/app/documentation/component-docs/button-split/examples/button-split-examples.component.ts]: add a empty line between the components

File: docs/app/documentation/component-docs/button-split/examples/button-split-template-example.component.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<fd-button-split>
+    <ng-template fd-button-split-action-title>
+        <u>Action Underscored</u>
```
Comment (MattL75) [inline docs/app/documentation/component-docs/button-split/examples/button-split-template-example.component.html]: Better to make it more obvious that this is for complex action titles. Maybe adding an icon on the left side of the text instead?

File: library/src/lib/button-split/button-split-utils/button-split.directives.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+ *                       <a [routerLink]="'/'">option2</a>
+ *                    </li>
+ *                </ul>
+ *            </fd-menu>
+ *        </div>
+ *    </fd-button-split>
+ *</fd-button-split>
+ *```
+ */
+
```
Comment (MattL75) [inline library/src/lib/button-split/button-split-utils/button-split.directives.ts]: remove empty line between the comment and directive annotation

File: library/src/lib/button-split/button-split.component.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+ *    </fd-button-split>
+ * ```
+ */
+@Component({
+    // TODO to be discussed
+    // tslint:disable-next-line:directive-selector
+    selector: 'fd-button-split',
+    templateUrl: 'button-split.component.html'
+})
+export class ButtonSplitComponent extends AbstractFdNgxClass {
```
Comment (MattL75) [inline library/src/lib/button-split/button-split.component.ts]: If it's not needed, better to not extend AbstractFdNgxClass. I'm currently looking to progressively phase this out as we go

**Comment by mikerodonnell89** [comment]
I think the width of the popover body should match the width of the popover control

**Comment by JKMarkowski** [comment]
@MattL75  I changed the component naming which unfortunately caused all of yours comments outdated. About the accessibility, I went through it with screen reader and it worked same as it is on our current popover/dropdown.

@mikerodonnell89 I added default popover body matching width of popover control. It can be disabled anytime by user.

**Comment by JKMarkowski** [comment]
Sure I removed links and added some alerts on click. I also think that it's not great for user, when he is redirected to home page after clicking on link. We should change it on other components.

**Comment by MattL75** [comment]
@JKMarkowski Yeah I agree, thanks for changing it

File: docs/app/documentation/component-docs/split-button/split-button-docs.component.html [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+</description>
+<component-example [name]="'ex3'">
+  <fd-split-button-icons-example></fd-split-button-icons-example>
+</component-example>
+<code-example [code]="buttonSplitIcons" [language]="'HTML'"></code-example>
+
+<separator></separator>
+<h2>Open/Close programmatically</h2>
+<description>
+    There is possibility to control open/close of popover by 2-ways data binding <code>[(isOpen)]</code>
```
Comment (MattL75) [inline docs/app/documentation/component-docs/split-button/split-button-docs.component.html]: It is possible to control the open state of the popover through two-way binding to the isOpen property.

File: docs/app/documentation/component-docs/split-button/split-button-docs.component.html [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  Use
+  <code class="code-snippet">[glyph]="'{{ '{' }}icon-name}'"</code> to change an icon of dropdown menu.
+</description>
+<component-example [name]="'ex3'">
+  <fd-split-button-icons-example></fd-split-button-icons-example>
+</component-example>
+<code-example [code]="buttonSplitIcons" [language]="'HTML'"></code-example>
+
+<separator></separator>
+<h2>Open/Close programmatically</h2>
```
Comment (MattL75) [inline docs/app/documentation/component-docs/split-button/split-button-docs.component.html]: Open/Close Programatically

(just missing capital P in programatically)

File: docs/app/documentation/component-docs/split-button/split-button-docs.component.html [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+<description>
+    There is possibility to control open/close of popover by 2-ways data binding <code>[(isOpen)]</code>
+</description>
+<component-example [name]="'ex4'">
+    <fd-split-button-programmatical-example></fd-split-button-programmatical-example>
+</component-example>
+<code-example [code]="buttonSplitProgrammaticallyExample" [language]="'HTML'"></code-example>
+
+<separator></separator>
+<h2>Adding template to action button</h2>
```
Comment (MattL75) [inline docs/app/documentation/component-docs/split-button/split-button-docs.component.html]: Custom Template for Action Button

File: docs/app/documentation/component-docs/split-button/split-button-docs.component.html [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+<description>
+    There is possibility to add some html template to action button with <code>fd-split-button-action-title</code> directive
```
Comment (MattL75) [inline docs/app/documentation/component-docs/split-button/split-button-docs.component.html]: It is possible to add a custom template to the action button through the <code>fd-split-button-action-title</code> directive.

File: docs/app/documentation/component-docs/split-button/split-button-header/split-button-header.component.html [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<header>Split Button</header>
+<description>
+    The Split Button component is made by connecting popover component and button directive.
```
Comment (MattL75) [inline docs/app/documentation/component-docs/split-button/split-button-header/split-button-header.component.html]: The split button component is composed of the popover and button components.

**Comment by JKMarkowski** [comment]
All of these changed are resolved now. Thanks for checking it, I really should focus more on documentation, instead of rushing.

**Comment by mikerodonnell89** [comment]
It does need keyboard support for selecting options.  Refer to the combobox component keydown handlers for examples on how this should work

**Comment by JKMarkowski** [comment]
Hi @mikerodonnell89 , @MattL75 I am a little confused, because for dropdown/popover/default menu component we do not support arrowup/arrowdown navigation, I was more thinking about implementing some keyboard handler functionality on menu component. A lot of components are based on this menu component and it would provide functionallity to all of them. What do you think about it ?

**Pull request merged**